safe/gui/tools/options_dialog.py

```diff
@@ -107,48 +106,89 @@ def __init__(self, iface, dock=None, parent=None, qsetting=''):
             self.set_org_disclaimer)
 
     def restore_state(self):
-        """Reinstate the options based on the user's stored session info.
-        """
-        # flag = settings.value(
-        #     'inasafe/useThreadingFlag', False)
-        # hack set use thread to false see #557
-        # TODO: Store default values somewhere else, e.g. defaults.py
+        """Reinstate the options based on the user's stored session info."""
         flag = False
         self.cbxUseThread.setChecked(flag)
```

Comment:
Historical:
There is a bug when the user enable the `use thread` option. So, we set it to alwasy False and hide the check box from the options dialog to prevent the user set it accidentally. 
Perhaps, there is still possibility to support it in the future.